File: dotNet/astenn/astenn-core/src/Org/Lestr/Astenn/Plugin/IServer.cs

using System;

namespace Org.Lestr.Astenn.Plugin
{
	
	
	public interface IServer
	{


        string Name { get;  }


        void Start();


        void Stop();


        string GetPluginRemoteAddress(Type pluginInterfaceClass, Type pluginImplementationClass);


    }// END Interface IServer


}// END Namespace Org.Lestr.Astenn.Plugin

File: tags/astenn-1.0.0/astenn-core/src/main/java/org/lestr/astenn/plugin/IServer.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.lestr.astenn.plugin;

/**
 *
 * @author pibonnin
 */
public interface IServer {


    String getName();


    void start() throws Exception;


    void stop() throws Exception;


    String getPluginRemoteAddress(Class<?> pluginInterfaceClass, Class<?> pluginImplementationClass);
    

}// END Interface IServer
